pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -257,8 +260,12 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 		})
 	}
 
+	httpProxyPort := mesh.ProxyHttpPort
+	if httpProxyPort == 0 && noneMode { // make sure http proxy is enabled for 'none' interception.
+		httpProxyPort = int32(pilot.DefaultPortHTTPProxy)
```

Comment:
but why? shouldn't user have control over declaring (or not declaring http proxy)?

Reply:
Also the alternative is to set the mesh port - which enable http proxy for both REDIRECT and NONE. 

This may be a good thing actually, and we could change the default install to have it on by default ( we already enable 4 ports by default ). It may provide benefits for some users. I'm happy to do the change in separate PR if we decide to go with that.